### DISCUSSION TITLE
How to import class BaseTestCase and use at main.py

### DISCUSSION THREAD

**Discussion by TuDoan95** [seleniumbase/SeleniumBase#2438] [Q&A] [answered]
from seleniumbase import BaseCase

url = 'https://www.google.com/'
class BaseTestCase(BaseCase):
    def setUp(self):
        super().setUp()
        # <<< Run custom setUp() code for tests AFTER the super().setUp() >>>

    def tearDown(self):
        self.save_teardown_screenshot()  # If test fails, or if "--screenshot"
        if self.has_exception():
            # <<< Run custom code if the test failed. >>>
            pass
        else:
            # <<< Run custom code if the test passed. >>>
            pass
        # (Wrap unreliable tearDown() code in a try/except block.)
        # <<< Run custom tearDown() code BEFORE the super().tearDown() >>>
        super().tearDown()

    def login(self):
        self.get(url)
        self.sleep(5)
        # <<< Placeholder. Add your code here. >>>
        # Reduce duplicate code in tests by having reusable methods like this.
        # If the UI changes, the fix can be applied in one place.
        pass

    def example_method(self):
        # <<< Placeholder. Add your code here. >>>
        pass
        
        -----------------------------------------------------------------------
        from events_seleniumbase.sb import BaseTestCase

if __name__ == "__main__":
    BaseTestCase().login()

**Comment by mdmintz** [comment, marked as answer]
See [base_test_case.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/boilerplates/base_test_case.py) and [boilerplate_test.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/boilerplates/boilerplate_test.py)
